### DISCUSSION TITLE
Add a more custom options to the image

### DISCUSSION THREAD

**Discussion by anjitpariyar** [dromru/react-photoswipe-gallery#924] [General]
I see the height and width options are available. but I find it hard to add more details, like object-fit, and the original size of the image. and if I add it with CSS, transition broke down

**Comment by akhmadullin** [comment]
Hi.

`width` and `height` [props](https://github.com/dromru/react-photoswipe-gallery#item) of `Item` component are responsible for informing PhotoSwipe about original size of the image, that will be opened in gallery.

Our lib is a React wrapper around [PhotoSwipe](https://github.com/dimsemenov/PhotoSwipe).

PhotoSwipe does not provide API to set `object-fit` value to image opened in gallery, so we can not implement this feature in our lib.

As a possible solution, you can 
1. create image component with all CSS properties, that you need
2. use it as [custom slide content](https://github.com/dromru/react-photoswipe-gallery#custom-slide-content) instead of setting `src` prop to Item component

But in this case there will be no zoom transition during opening gallery after user's click on thumbnail.
